openmc/cell.py

```diff
@@ -229,7 +230,7 @@ def bounding_box(self):
 
     @property
     def num_instances(self):
-        return len(self.paths)
+        return self._num_instances
```

Comment:
In the old code, the act of checking ``len(self.paths)`` would trigger an exception if ``self._paths`` was either  ``[]`` or ``None``.